client/components/Recipe.js

```diff
@@ -48,10 +48,17 @@ class Recipe extends React.Component {
     this.props.wholeRecipeDispatch(this.props.data.audio_path)
   }
 
-  checkReady(){
-    const { playing } = this.props.data
-    if (playing !== undefined) {
-      return <Audio currentStep={this.props.data.currentStep} audio_path={this.props.data.audio_path[this.props.data.currentStep - 1]} playing={playing}/>
+  checkAudioReady(){
+    const { playing, currentStep, audio_path, whole_recipe_ingredients_audio_path } = this.props.data
+
+    // This case for an array of paths
+    if (audio_path !== undefined && whole_recipe_ingredients_audio_path === '') {
+      return <Audio currentStep={currentStep} audio_path={audio_path[currentStep - 1]} playing={playing}/>
+    }
+
+    // Here is for the whole or ingredients audio path
+     else if (whole_recipe_ingredients_audio_path !== undefined){
+      return <Audio currentStep={currentStep} audio_path={whole_recipe_ingredients_audio_path} playing={playing}/>
```

Comment:
state you might like to track is : 
- which audio flow you're in
  1. ingredients
  2. whole recipe
  3. stepped recipe
     - which step ?